src/disable-amp.js

```diff
@@ -44,4 +44,10 @@ function preventAmp() {
 if (document.location.origin.includes('.google.')) {
     preventAmp();
     observeDomChanges(preventAmp);
+} else {
+    // Redirects "/amp/" website to a stripped versionconst
+    persistentURL = document.URL;
+    if (persistentURL.endsWith("/amp/")) {
```

Comment:
we cannot add `?nonamp=1` to all addresses finished with `/amp`. it potentially could brake some websites. is it possible to make stronger check for this case?

Reply:
There would be a solution, but unfortunately I don't know how to apply it.

It would be enough to verify that website gives status 301, in addition to finished with "/amp/".

Reply:
I rewrote the code to find the element containing the link to remove amp (the one ending with "?noamp=1"). Now every website that doesn't contain this element will not be touched.

https://github.com/AdguardTeam/DisableAMP/pull/21/commits/02a1d60e5400cf7d2277f955eb6977e614b9dc6b

Reply:
1. it can be simplified to
`const redirectURL = document.querySelector('a[href$="?nonamp=1"]');`
`if (redirectURL) {
    document.location.replace(redirectURL);
}`
2. it's also unsafe.

Reply:
Unsafe?

Reply:
P.S: This is the fix based on your suggestion https://github.com/AdguardTeam/DisableAMP/pull/21/commits/820f6d9980143af02550cdffc8bf9773bd236687

Reply:
1. On a page with some random links ending with `?nonamp=1`, you will replace current document location with the `href` of the first link - is it really what you wanna do?

2. I guess searching for these links you are trying to detect that you are on amp page - but as I see this is not correct. This could work in one special case, but for sure it's not sufficiently enough to work correctly on any website.

Reply:
My intention was to prevent the site redirecting to the persistent amp version, and for doing this I definitively chose to verify that there was an element containing a link with tag to deactivate this amp page (a page without this link inside an element can't be an AMP page I guess). In any case I agree that not all sites use this exact tag to switch to standard page, but at the moment windowsteca.net is my only website reference (in absence of a unique solution for every website, that I don't know at the moment).